pytest_doctestplus/plugin.py

```diff
@@ -208,25 +207,34 @@ def collect(self):
                     yield doctest_plugin.DoctestItem(
                         test.name, self, runner, test)
 
-    class DocTestTextfilePlus(doctest_plugin.DoctestItem, pytest.Module):
-        # Some pytest plugins such as hypothesis try and access the 'obj'
-        # attribute, and by default this returns an error for this class
-        # so we override it here to avoid any issues.
-        def obj(self):
-            pass
-
-        def runtest(self):
-            # satisfy `FixtureRequest` constructor...
-            self.funcargs = {}
-            fixture_request = doctest_plugin._setup_fixtures(self)
-
-            options = get_optionflags(self) | FIX
+    class DocTestTextfilePlus(doctest_plugin.DoctestTextfile):
 
-            doctest.testfile(
-                str(self.fspath), module_relative=False,
-                optionflags=options, parser=DocTestParserPlus(),
-                extraglobs=dict(getfixture=fixture_request.getfixturevalue),
-                raise_on_error=True, verbose=False, encoding='utf-8')
+        def collect(self):
+            from _pytest.doctest import (_get_runner, _get_checker,
+                                         _get_continue_on_failure, DoctestItem)
+
+            encoding = 'utf-8'
+            text = self.fspath.read_text(encoding)
+            filename = str(self.fspath)
+            name = self.fspath.basename
+            globs = {"__name__": "__main__"}
+
+            optionflags = get_optionflags(self) | FIX
+
+            runner = _get_runner(
+                verbose=False,
+                optionflags=optionflags,
+                checker=_get_checker(),
+                continue_on_failure=_get_continue_on_failure(self.config))
+
+            parser = DocTestParserPlus()
+            test = parser.get_doctest(text, globs, name, filename, 0)
+            if test.examples:
+                if PYTEST_LT_5_4:
+                    yield DoctestItem(test.name, self, runner, test)
+                else:
+                    yield DoctestItem.from_parent(
```

Comment:
This does not work because I think now Python's `doctest` tries to read in the docstring as it is defined in the test file, not via `testdir` fixture. I stared at the code here, the code in `pytest`, and Python doc, but couldn't figure out a path forward.

Simply renaming the old `runtest` to `collect` doesn't work either:

https://github.com/astropy/pytest-doctestplus/blob/9975e23d178465c00b75f6b2f178f89e2245b540/pytest_doctestplus/plugin.py#L218